software/17.gRPC/README.md

```diff
@@ -1,144 +1,264 @@
+- Create a `New` function, which creates a new `Client` instance
+- The `New` function initialize two members of the structure
+  
+  - `conn` which is a `grpc.ClientConn`
+  - `core` which is your messenger's client
+  - The connection can be insecure for now
+  
+- Create a `Disconnect` method, which closes the client connection
+- Create a `Send` method, which sends the message given as parameter. It returns the response received from the server
 
+In the `main.go` file:
 
-## :microscope: Step 5 - Adding some tests
+- Below the print of `Starting the client`, create a new client and send a message
 
-> :bulb: Well writing code is nice, but without adding some tests ? Hell no.
+### :books: **Documentation**:
 
-In the `tests/chat.test.ts`, you'll write tests to ensure that the communication between the client and the server works properly.
+- Look at the protobuf generated file :)
 
-Remember to start the server with the following command whenever you run the client tests:
+### :heavy_check_mark: **Validation**:
+- Run the server
 ```shell
-yarn start:server
+go run main.go "-s"
+```
+- Open a new terminal, and run
+```shell
+go run main.go "-c"
 ```
 
-Add the following tests to the suite using [jest](https://jestjs.io/):
+The server should print:
+```shell
+$ go run main.go "-s"
+Starting the server
+[SERVER]: received Hello World! at 1628075261
+```
+The message and the timestamp are obviously different from mine.
 
-- It can create the client
+## Step 5 - Shutdown properly
 
-- It can send the hello message
+### :bookmark_tabs: **Description**:
+When you don't need the server to run anymore, or whenever the client has stopped sending their requests, when need 
+to shut it down properly.
 
-- The server can respond `hello from server` to any incoming `hello` gRPC call.
+As you created the `Disconnect` method for the client, you only need to create the same method for the server that 
+triggers whenever you hit `CTRL + C`.
 
-> :bulb: You'll have to modify the implementation of the service in the server and use the second argument.
+### :pushpin: **Tasks**:
+- Capture when the user hit `CTRL + C` on the server-side
+- Create a `Stop` method for the server and call it when necessary
+- Use the `Disconnect` method for the client when it has sent its message
 
-##### Resources
+### :books: **Documentation**:
+- [Signals in go](https://pkg.go.dev/os/signal)
 
-- [Test suite with Jest](https://jestjs.io/docs/api)
+### :heavy_check_mark: **Validation**:
+Try to hit `CTRL + C` while running the server, and it should shut down.
 
-## :rocket: To go further
+## To go further
 
-If you want to learn more about gRPC project out there, check out [this](https://github.com/grpc-ecosystem/awesome-grpc) list of amazing project that uses it.
+For the moment, you are using an int to represent the date, but there is the [Timestamp](https://stackoverflow.
+com/questions/3574716/date-and-time-type-for-use-with-protobuf) type available in protobuf.
```

Comment:
You should also add your old bonus

```
- If you want to learn more about gRPC project out there, check out [this](https://github.com/grpc-ecosystem/awesome-grpc) list of amazing projects that uses it.
- You can add more services to the protobuf file, passing other type of parameters instead of a string, or even combining several data types.
```

Reply:
Nice catch ;)